ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Monitor.hs

```diff
@@ -122,7 +118,7 @@ inboundPeers PeerSelectionActions{
     return $ \_ ->
       let -- If peer happens to already be present in the Known Peer set
           -- 'insert' is going to do its due diligence before adding.
-          newEntry    = Map.singleton addr (ps, DoAdvertisePeer, IsNotLedgerPeer)
+          newEntry    = Map.singleton addr (Just ps, Nothing, Nothing)
```

Comment:
Actually when the connection manager writes to the `infoChannel` the connection is already negotiated.  So we should know the advertise flag by now.

Reply:
https://github.com/input-output-hk/ouroboros-network/issues/4647